### PR TITLE
Adding impl/test crumbs for requirements

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

Adding impl/test crumbs for assorted requirements.

## Why is the change being made?

This is part of on-going requirements work.

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [ ] No [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were altered.
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/reactor/blueprints/componentBlueprint.py [hunk 1]
```diff
@@ -291,6 +291,10 @@ def insertDepletableNuclideKeys(c, blueprint):
     """
     Auto update number density keys on all DEPLETABLE components.
 
+    .. impl:: Any depletable component will be adjusted.
+        :id: I_ARMI_BP_NUC_FLAGS
+        :implements: R_ARMI_BP_NUC_FLAGS
+
```
[the remark this answers is not in the document]
Comment (keckler) [reply armi/reactor/blueprints/componentBlueprint.py:297]: Hm, or maybe this is just one part of it. I guess the user needs to define composite objects with the `DEPLETABLE` flag, which would happen in other parts of the blueprints.

File: armi/physics/neutronics/globalFlux/globalFluxInterface.py [hunk 2]
```diff
@@ -119,8 +119,13 @@ def interactEOC(self, cycle=None):
                 * units.ABS_REACTIVITY_TO_PCM
             )
 
-    def _checkEnergyBalance(self):
-        """Check that there is energy balance between the power generated and the specified power is the system."""
+    def checkEnergyBalance(self):
```
[the remark this answers is not in the document]
Comment (john-science) [reply armi/physics/neutronics/globalFlux/globalFluxInterface.py:122]: I don't think that's a subclass or anything. Just a similar name.

Tell me if I'm wrong on that.

File: armi/reactor/blueprints/componentBlueprint.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (john-science) [reply armi/reactor/blueprints/componentBlueprint.py:297]: This is just a "blueprint requirement", so I put it in the most general place in the blueprints.

That's really as far as my thought process went.

Hmmm

File: armi/physics/neutronics/globalFlux/globalFluxInterface.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (keckler) [reply armi/physics/neutronics/globalFlux/globalFluxInterface.py:122]: I think you're right.

File: armi/reactor/blueprints/componentBlueprint.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (keckler) [reply armi/reactor/blueprints/componentBlueprint.py:297]: Okay, well it is good where it is, by the logic of my second comment above. But during the upcoming "real" review, I anticipate we'll probably ask for another impl in the `isotopicOptions`.

[the remark this answers is not in the document]
Comment (john-science) [reply armi/reactor/blueprints/componentBlueprint.py]: Updated. Take a look.

**Review by keckler — APPROVED** [review]
Thanks for the additions!

**Pull request merged**